### PR TITLE
Archive video applied

### PR DESCRIPTION
**Pull request by ShadyNagy**
- Archive video used.
- Uploader read the encoded date from the file.

### CODE REVIEW

File: src/DevBetterWeb.Core/Entities/Member.cs [hunk 1]
```diff
@@ -57,7 +57,19 @@ internal Member(string userId)
     public decimal? CityLatitude { get; set; }
     public decimal? CityLongitude { get; set; }
     public List<BillingActivity> BillingActivities { get; set; } = new List<BillingActivity>();
+    public List<MemberVideo> Videos { get; private set; } = new List<MemberVideo>();
 
+    public void AddVideo(MemberVideo video)
```
Comment (ardalis) [inline src/DevBetterWeb.Core/Entities/Member.cs:62]: They're not actually adding a video, but are adding some stats about videos they've watched, right?

File: src/DevBetterWeb.Core/Entities/MemberVideo.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using DevBetterWeb.Core.Interfaces;
+using DevBetterWeb.Core.SharedKernel;
+
+namespace DevBetterWeb.Core.Entities
+{
+  public class MemberVideo : BaseEntity, IAggregateRoot
```
Comment (ardalis) [inline src/DevBetterWeb.Core/Entities/MemberVideo.cs:6]: Maybe renamed to `MemberVideoProgress` ?

**Review by ardalis — APPROVED** [review]

File: src/DevBetterWeb.Core/Entities/MemberVideo.cs (hunk 2 above)
Comment (ShadyNagy) [reply src/DevBetterWeb.Core/Entities/MemberVideo.cs]: Yes, that is better.
Done

File: src/DevBetterWeb.Core/Entities/Member.cs (hunk 1 above)
Comment (ShadyNagy) [reply src/DevBetterWeb.Core/Entities/Member.cs]: changed to AddVideoProgress

**Comment by ardalis** [comment]
Let's see how it works...

**Pull request merged**